src/System.Numerics.Vectors/tests/Performance/Constructor/GenericVectorConstructorTests.cs

```diff
@@ -0,0 +1,353 @@
+        {
+            UInt64[] arrValues = GenerateRandomValuesForVector<UInt64>();
+            var spanValues = new ReadOnlySpan<UInt64>(arrValues);
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    SpanCast<UInt64>(spanValues);
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = DefaultInnerIterationsCount)]
+        public static void SpanCastBenchmark_Int64()
+        {
+            Int64[] arrValues = GenerateRandomValuesForVector<Int64>();
+            var spanValues = new ReadOnlySpan<Int64>(arrValues);
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    SpanCast<Int64>(spanValues);
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = DefaultInnerIterationsCount)]
+        public static void SpanCastBenchmark_Single()
+        {
+            Single[] arrValues = GenerateRandomValuesForVector<Single>();
+            var spanValues = new ReadOnlySpan<Single>(arrValues);
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    SpanCast<Single>(spanValues);
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = DefaultInnerIterationsCount)]
+        public static void SpanCastBenchmark_Double()
+        {
+            Double[] arrValues = GenerateRandomValuesForVector<Double>();
+            var spanValues = new ReadOnlySpan<Double>(arrValues);
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    SpanCast<Double>(spanValues);
+                }
+            }
+        }
+
+
+        public static void SpanCast<T>(ReadOnlySpan<T> values) where T : struct
+        {
+            for (var iteration = 0; iteration < Benchmark.InnerIterationCount; iteration++)
+            {
+                ReadOnlySpan<Vector<T>> vectors = MemoryMarshal.Cast<T, Vector<T>>(values);
+                Vector<T> vector = vectors[0];
+            }
+        }
+
+        internal static T[] GenerateRandomValuesForVector<T>() where T : struct
+        {
+            int minValue = GetMinValue<T>();
+            int maxValue = GetMaxValue<T>();
+            return Util.GenerateRandomValues<T>(Vector<T>.Count, minValue, maxValue);
+        }
+
+        internal static int GetMinValue<T>() where T : struct
+        {
+            if (typeof(T) == typeof(Int64) || typeof(T) == typeof(Single) || typeof(T) == typeof(Double) || typeof(T) == typeof(UInt32) || typeof(T) == typeof(UInt64))
+            {
+                return int.MinValue;
+            }
+            var typeInfo = typeof(T).GetTypeInfo();
```

Comment:
nit: don't use var here and below (only use it where the type can be statically inferred from the right hand side).

Reply:
Actually all this has been copy pasted from the normal test code [here](https://github.com/WinCPP/corefx/blob/1da32ab8ab887128ae3056f0590fd2a5c211f6e2/src/System.Numerics.Vectors/tests/GenericVectorTests.tt#L1802-L1811)

Reply:
Done.